src/Databases/DatabaseFactory.cpp

```diff
@@ -432,6 +433,24 @@ DatabasePtr DatabaseFactory::getImpl(const ASTCreateQuery & create, const String
         return std::make_shared<DatabaseSQLite>(context, engine_define, create.attach, database_path);
     }
 #endif
+    else if (engine_name == "Filesystem")
+    {
+        const ASTFunction * engine = engine_define->engine;
+
+        /// If init_path is empty, then the current path will be used
+        std::string init_path;
+
+        if (engine->arguments && !engine->arguments->children.empty())
+        {
+            if (engine->arguments->children.size() != 1)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Filesystem database requires exactly 1 argument: filesystem_path");
```

Comment:
seems the previous "at most 1" was correct, I did not notice that zero arguments is allowed